### PR TITLE
Move enum pe_obj_types to libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**
This is part of a project to merge the functionality of libpe_status and libpe_rules into libcrmcommon, bringing it up to current guidelines in the process

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , can you review this when you get a chance? Low priority, thanks

**Comment by kgaillot** [comment]
Accidentally pushed to wrong branch, it's back to the original commits now

**Comment by kgaillot** [comment]
retest this please

File: include/crm/common/resources.h [hunk 1]
```diff
@@ -27,6 +27,7 @@ enum pe_obj_types {
     pcmk_rsc_variant_primitive  = 0,    //!< Primitive resource
     pcmk_rsc_variant_group      = 1,    //!< Group resource
     pcmk_rsc_variant_clone      = 2,    //!< Clone resource
+    pcmk_rsc_variant_bundle     = 3,    //!< Bundle resource
```
Comment (wenningerk) [inline include/crm/common/resources.h:30]: for consistency we might use that for pe_container below - but well ...

File: lib/pacemaker/pcmk_sched_actions.c [hunk 2]
```diff
@@ -144,7 +144,8 @@ action_uuid_for_ordering(const char *first_uuid, const pe_resource_t *first_rsc)
         if (pcmk_is_set(first_rsc->flags, pe_rsc_notify)
             && ((first_rsc->parent == NULL)
                 || (pe_rsc_is_clone(first_rsc)
-                    && (first_rsc->parent->variant == pe_container)))) {
+                    && (first_rsc->parent->variant == pcmk_rsc_variant_bundle))
+               )) {
```
Comment (wenningerk) [inline lib/pacemaker/pcmk_sched_actions.c:148]: do we state that explicitly in the guidelines? both alignment and maybe as well
when we decide not to close everything in the last line ... 
at least for my understanding this should be aligned with the block it closes.

File: tools/crm_resource_runtime.c [hunk 3]
```diff
@@ -2041,7 +2041,7 @@ cli_resource_execute(pe_resource_t *rsc, const char *requested_name,
     if (rsc->variant == pcmk_rsc_variant_group) {
         out->err(out, "Sorry, the %s option doesn't support group resources", rsc_action);
         return CRM_EX_UNIMPLEMENT_FEATURE;
-    } else if (rsc->variant == pe_container || pe_rsc_is_bundled(rsc)) {
+    } else if (pe_rsc_is_bundled(rsc)) {
```
Comment (wenningerk) [inline tools/crm_resource_runtime.c:2044]: guess the behavioral change here is rather a fix or simply irrelevant

**Comment by wenningerk** [comment]
Looking good! Just a slight inconsistency in the split of the patches in one place. A formatting question and a behavioral question that is probably irrelevent.

File: include/crm/common/resources.h (hunk 1 above)
Comment (kgaillot) [reply include/crm/common/resources.h:30]: whoops, i meant to

File: lib/pacemaker/pcmk_sched_actions.c (hunk 2 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:148]: no, this was just a really hard to follow condition. i think it'll be better to separate a bool variable for part of the condition

Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:148]: i believe if pe_rsc_notify is set and parent is NULL, that first_rsc has to be a clone, so we can move pe_rsc_is_clone(first_rsc) to the outer conditions

Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:148]: and at that point, parent == NULL or variant_bundle is redundant, since a clone has to meet one of those two conditions

File: tools/crm_resource_runtime.c (hunk 3 above)
Comment (kgaillot) [reply tools/crm_resource_runtime.c:2044]: it's irrelevant -- pe_rsc_is_bundled() will return true if rsc->variant is bundle

File: lib/pacemaker/pcmk_sched_actions.c (hunk 2 above)
Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:148]: apparently the bundle resource itself can have pe_rsc_notify set and hit parent==NULL, so a bool variable it will be

Comment (kgaillot) [reply lib/pacemaker/pcmk_sched_actions.c:148]: pe_rsc_is_clone(first_rsc) || pe_rsc_is_bundled(first_rsc)) doesn't break any regression tests, so I think that covers it

**Comment by kgaillot** [comment]
Updated last two commits per review

File: tools/crm_resource_runtime.c (hunk 3 above)
Comment (wenningerk) [reply tools/crm_resource_runtime.c:2044]: well it will move to the upmost parent first and then check variant ... but probably still irrelevant ...

Comment (kgaillot) [reply tools/crm_resource_runtime.c:2044]: Yep, bundles always have parent==NULL
